libs/threading/tests/unit/condition_variable2.cpp

```diff
@@ -379,7 +379,10 @@ template <typename Dur>
 void test_timed_wait(bool call_notify, bool call_interrupt, Dur dur)
 {
     // test the basic jthread API
-    bool ready = false;
+
+    // ready_ instead of ready to not clash with state::ready and work around
+    // potential bug in GCC 10
+    bool ready_ = false;
```

Comment:
This was a bit of an odd one, but it looks like it's a bug in the new GCC 10. It ICE's with this: https://cdash.cscs.ch//viewBuildError.php?buildid=108701. I couldn't reproduce it with a very small example, so I'm just going to put this as a temporary solution. If that link dies the message is:

```
../../libs/threading/tests/unit/condition_variable2.cpp: In function âvoid test_timed_wait(bool, bool, Dur)â:                                                                                                                                                                                               
../../libs/threading/tests/unit/condition_variable2.cpp:389:9: internal compiler error: Segmentation fault                                                                                                                                                                                                  
  389 |         ready,                                                                                                                                                                                                                                                                                      
      |         ^~~~~                                                                                                                                                                                                                                                                                       
0xc95cef crash_signal                                                                                                                                                                                                                                                                                       
        ../../gcc-10.1.0/gcc/toplev.c:328                                                                                                                                                                                                                                                                   
0x7e363b comptypes(tree_node*, tree_node*, int)                                                                                                                                                                                                                                                             
        ../../gcc-10.1.0/gcc/cp/typeck.c:1505                                                                                                                                                                                                                                                               
0x737aca check_local_shadow                                                                                                                                                                                                                                                                                 
        ../../gcc-10.1.0/gcc/cp/name-lookup.c:2765                                                                                                                                                                                                                                                          
0x737aca do_pushdecl                                                                                                                                                                                                                                                                                        
        ../../gcc-10.1.0/gcc/cp/name-lookup.c:3112                                                                                                                                                                                                                                                          
0x738010 pushdecl(tree_node*, bool)                                                                                                                                                                                                                                                                         
        ../../gcc-10.1.0/gcc/cp/name-lookup.c:3177                                                                                                                                                                                                                                                          
0x6e82b3 build_enumerator(tree_node*, tree_node*, tree_node*, tree_node*, unsigned int)                                                                                                                                                                                                                     
        ../../gcc-10.1.0/gcc/cp/decl.c:15892                                                                                                                                                                                                                                                                
0x753947 cp_parser_enumerator_definition                                                                                                                                                                                                                                                                    
        ../../gcc-10.1.0/gcc/cp/parser.c:19493                                                                                                                                                                                                                                                              
0x753947 cp_parser_enumerator_list                                                                                                                                                                                                                                                                          
        ../../gcc-10.1.0/gcc/cp/parser.c:19422                             
0x753947 cp_parser_enum_specifier                                          
        ../../gcc-10.1.0/gcc/cp/parser.c:19353                             
0x753947 cp_parser_type_specifier                                          
        ../../gcc-10.1.0/gcc/cp/parser.c:17683                             
0x753f46 cp_parser_decl_specifier_seq                                      
        ../../gcc-10.1.0/gcc/cp/parser.c:14359                             
0x754921 cp_parser_simple_declaration                                      
        ../../gcc-10.1.0/gcc/cp/parser.c:13613                             
0x75660a cp_parser_declaration_statement                                   
        ../../gcc-10.1.0/gcc/cp/parser.c:13166                             
0x7571e5 cp_parser_statement                                               
        ../../gcc-10.1.0/gcc/cp/parser.c:11472                             
0x757ff8 cp_parser_statement_seq_opt                                       
        ../../gcc-10.1.0/gcc/cp/parser.c:11838                             
0x7580b0 cp_parser_compound_statement                                      
        ../../gcc-10.1.0/gcc/cp/parser.c:11788                             
0x76db70 cp_parser_function_body                                           
        ../../gcc-10.1.0/gcc/cp/parser.c:23063                             
0x76db70 cp_parser_ctor_initializer_opt_and_function_body                                                                                             
        ../../gcc-10.1.0/gcc/cp/parser.c:23114                             
0x77117d cp_parser_function_definition_after_declarator                                                                                               
        ../../gcc-10.1.0/gcc/cp/parser.c:28996                             
0x77205f cp_parser_function_definition_from_specifiers_and_declarator                                                                                 
        ../../gcc-10.1.0/gcc/cp/parser.c:28912                             
Please submit a full bug report,                                           
with preprocessed source if appropriate.                                   
Please include the complete backtrace with any bug report.                                                                                            
See <https://gcc.gnu.org/bugs/> for instructions.
```